shared/search/providers.ts

```diff
@@ -570,3 +631,26 @@ export async function raceWithDelayOffset<T>(
 async function delay(timeout: number): Promise<undefined> {
     return new Promise(r => setTimeout(r, timeout))
 }
+
+/**
+ * Adds options to include forked and archived repositories.
+ *
+ * @param queryTerms The terms of the search query.
+ * @param includeFork Whether or not the include forked repositories regardless of settings.
+ * @param includeArchived Whether or not the include archived repositories regardless of settings.
+ */
+function addRepositoryKindTerms(
+    queryTerms: string[],
+    includeFork: boolean,
+    includeArchived: boolean
+): string[] {
+    if (includeFork || getConfig('basicCodeIntel.includeForks', false)) {
+        queryTerms.push('fork:yes')
```

Comment:
The API is a little odd here in that we are modifying the input argument but also returning it. The usage site makes it look like we're getting a copy, but we're not really—the original has been tweaked.

I'd just let it modify in-place and document it as such, probably.

Reply:
Yeah that was dirty. Updated.